### ISSUE TITLE
question regarding clustering evaluation in MTEB

### ISSUE THREAD

**Issue by OnAnd0n** [embeddings-benchmark/mteb#2268]
Hello,
I am some interested in evaluation on Korean text embeddings.

I have a question about the clustering task evaluation method in MTEB.

MTEB evaluates clustering using the K-means algorithm. 
However, since K-means measures distance based on Euclidean space, 
I think this approach may not be suitable for text embedding models trained to measure similarity using cosine similarity or inner product.

so, I considered an alternative approach that applies spectral clustering on a graph constructed using cosine similarity.

In fact, I observed an improvement in the V-measure score with this method.

I would like to hear your thoughts on this approach.

Best regards,

`import networkx as nx

class ClusteringEvaluator_REV(ClusteringEvaluator):
    def __call__(self, model: Encoder, *, encode_kwargs: dict[str, Any] = {}):
        if "batch_size" not in encode_kwargs:
            encode_kwargs["batch_size"] = 32

        corpus_embeddings = model.encode(
            self.sentences,
            task_name=self.task_name,
            **encode_kwargs,
        )
        G = nx.Graph()

        # logger.info("Fitting Faiss K-Means model...")
        for i, i_text in enumerate(self.sentences[:-1]):
          score_list = model.similarity(corpus_embeddings[i], corpus_embeddings[i+1:])[0]*100

          for j_text, score in zip(self.sentences[i+1:], score_list):
                G.add_edge(i_text, j_text, weight=score)

        adjacency_cos_score_matrix = nx.to_numpy_array(G)
        adjacency_cos_score_matrix = np.where(adjacency_cos_score_matrix < 0, 0, adjacency_cos_score_matrix)
        display(adjacency_cos_score_matrix)

        clustering = sklearn.cluster.SpectralClustering(n_clusters=len(set(self.labels)), affinity = 'precomputed', assign_labels='discretize')
        clustering.fit(adjacency_cos_score_matrix)
        cluster_assignment = clustering.labels_

        print(cluster_assignment)

        # logger.info("Evaluating...")
        v_measure = metrics.cluster.v_measure_score(self.labels, cluster_assignment)

        return {"v_measure": v_measure}`

|                                         |   k-means |   cos & spectral |
|:----------------------------------------|----------:|-----------------:|
| intfloat/multilingual-e5-large-instruct |     26.62 |            34.42 |
| BAAI/bge-m3                             |     33.01 |            47.82 |
| nlpai-lab/KoE5                          |     36.22 |            47.95 |
| dragonkue/BGE-m3-ko                     |     29.49 |            40    |
| upskyy/bge-m3-korean                    |     22.08 |            26.14 |
| nlpai-lab/KURE-v1                       |     39.21 |            53.46 |
| Snowflake/snowflake-arctic-embed-l-v2.0 |     40.68 |            47.53 |
